src/CheckSum.cs

```diff
@@ -16,6 +17,38 @@ public static class Checksum
     {
         public const uint Seed = 0xFFFFFFFF;
 
+        /// <summary>
+        /// Compute a CRC32 checksum of data read from a given stream.
+        /// </summary>
+        /// <remarks>the given <see cref="Stream"/>'s position will be set to the given offset.</remarks>
+        /// <param name="s"><see cref="Stream"/> that should be read.</param>
+        /// <param name="offset"> position of the <see cref="Stream"/> in order to read</param>
+        /// <param name="length"> byte length of data that should be read from the stream.
+        /// </param>
+        /// <returns>the CRC32 checksum of the </returns>
+        public static uint CRC32(Stream s, int offset, int length)
+        {
+            var increment = Math.Min(length, MaxReadBufferSize);
+            var buffer = new byte[increment];
+            var remainder = length;
+            s.Position = offset;
+
+            var count = -1;
+            var checksum = Seed;
+            while (increment != 0 && count != 0)
```

Comment:
The logic surrounding `increment` and `count` can be improved: 

 - Make the name `count` more explicit: `readBytes`
 - Make the variable local to the loop, instead of defining it before the loop
 - Move the `if (count == 0) throw` test right after the call to `s.Read`
 - There is no need to store `increment` from one iteration of the loop to the next, it is enough to call `s.Read(buffer, 0, Math.Min(remainder, buffer.Length)`
 - `var buffer = new byte[Math.Min(length, MaxReadBufferSize)]`